src/generators/dom/index.js

```diff
@@ -168,7 +169,7 @@ export default function dom ( parsed, source, options ) {
 	if ( templateProperties.oncreate ) {
 		builders.init.addBlock( deindent`
 			if ( options._root ) {
-				options._root._renderHooks.push({ fn: ${generator.alias( 'template' )}.oncreate, context: this });
+				options._root._renderHooks.push( ${generator.alias( 'template' )}.oncreate.bind( this ) );
```

Comment:
(obviously that only works if `oncreate` is declared inline — if it's a reference we'd still need to `.call` it)